sdk/appcenter/src/test/java/com/microsoft/appcenter/channel/DefaultChannelTest.java

```diff
@@ -1071,4 +1086,164 @@ public void scheduleLogsWithNewToken() {
+
+        /* Check values. */
+        verifyStatic(never());
+        SharedPreferencesManager.putLong(eq(START_TIMER_PREFIX + TEST_GROUP), any(long.class));
+        verifyStatic();
+        SharedPreferencesManager.remove(eq(START_TIMER_PREFIX + TEST_GROUP));
+        verify(mAppCenterHandler).postDelayed(any(Runnable.class), eq(mockInterval));
+    }
+
+    @Test
+    public void checkPendingLogsZeroMorePendingLog() {
+        long mockInterval = 10000;
+        long mockSystemTime = 50000L;
+        final long mockPrefTime = 1000L;
+
+        /* Mock static classes. */
+        mockStatic(System.class);
+        when(System.currentTimeMillis()).thenReturn(mockSystemTime);
+        when(SharedPreferencesManager.getLong(eq(START_TIMER_PREFIX + TEST_GROUP))).thenReturn(mockPrefTime);
+
+        /* Mock channel. */
+        Persistence mockPersistence = mock(Persistence.class);
+        when(mockPersistence.countLogs(TEST_GROUP)).thenReturn(0);
+        Channel.GroupListener mockListener = mock(Channel.GroupListener.class);
+        AppCenterIngestion mockIngestion = mock(AppCenterIngestion.class);
+        DefaultChannel channel = new DefaultChannel(mock(Context.class), UUIDUtils.randomUUID().toString(), mockPersistence, mockIngestion, mAppCenterHandler);
+        channel.addGroup(TEST_GROUP, 10, mockInterval, MAX_PARALLEL_BATCHES, mockIngestion, mockListener);
+
+        /* Check values. */
+        verify(mAppCenterHandler, never()).postDelayed(any(Runnable.class), anyLong());
+        verifyStatic();
+        SharedPreferencesManager.remove(eq(START_TIMER_PREFIX + TEST_GROUP));
+    }
+
+    @Test
+    public void checkPendingLogsPendingLogMoreZero() {
+        long mockInterval = 10000;
+        long mockSystemTime = 50000L;
+        final long mockPrefTime = 1000L;
+
+        /* Mock static classes. */
+        mockStatic(System.class);
+        when(System.currentTimeMillis()).thenReturn(mockSystemTime);
+        when(SharedPreferencesManager.getLong(eq(START_TIMER_PREFIX + TEST_GROUP))).thenReturn(mockPrefTime);
+
+        /* Mock channel. */
+        Persistence mockPersistence = mock(Persistence.class);
+        when(mockPersistence.countLogs(TEST_GROUP)).thenReturn(5);
+        Channel.GroupListener mockListener = mock(Channel.GroupListener.class);
+        AppCenterIngestion mockIngestion = mock(AppCenterIngestion.class);
+        DefaultChannel channel = new DefaultChannel(mock(Context.class), UUIDUtils.randomUUID().toString(), mockPersistence, mockIngestion, mAppCenterHandler);
+        channel.addGroup(TEST_GROUP, 10, mockInterval, MAX_PARALLEL_BATCHES, mockIngestion, mockListener);
+
+        /* Check values. */
+        verify(mAppCenterHandler, never()).postDelayed(any(Runnable.class), anyLong());
+        verify(mockPersistence, times(2)).countLogs(eq(TEST_GROUP));
+    }
+
+    @Test
+    public void setAppSecretIngestionNoEquals() {
```

Comment:
I can't quite undestand what exactly this test is checking

Reply:
This test coverage the point, when `groupState.Ingestion != Ingestion` in setAppSecret method